pywr/control_curves.py

```diff
@@ -0,0 +1,76 @@
+"""
+This module contains a set of pywr._core.Parameter subclasses for defining control curve based parameters.
+"""
+
+from ._parameters import Parameter as BaseParameter
+import numpy as np
+
+
+class BaseParameterControlCurve(BaseParameter):
+    """ Base class for all Parameters that rely on a the attached Node containing a control_curve Parameter
+
+    """
+    def __init__(self, control_curve):
+        """
+
+        Parameters
+        ----------
+        control_curve : Parameter
+            The Parameter object to use as a control_curve. It should not be shared with other
```

Comment:
This is a potential pitfall. I imagine it not uncommon to want to share the same control curve between the cost and max_flow parameters. Perhaps this function should first check if `control_curve.parent` is already set and raise an exception if it is?